src/util/misc.rs

```diff
@@ -16,12 +16,184 @@
 //!
 //! Various utility functions
 
+use std::fmt;
+
+use secp256k1;
+use secp256k1::recovery::{RecoveryId, RecoverableSignature};
 use hashes::{sha256d, Hash, HashEngine};
+
 use blockdata::opcodes;
 use consensus::{encode, Encodable};
+use util::key::PublicKey;
+use util::address::{Address, AddressType};
 
 static MSG_SIGN_PREFIX: &[u8] = b"\x18Bitcoin Signed Message:\n";
 
+/// An error used for dealing with Bitcoin Signed Messages.
+#[derive(Debug, PartialEq, Eq)]
+pub enum MessageSignatureError {
+    /// Signature is expected to be 65 bytes.
+    InvalidLength,
+    /// The signature is invalidly constructed.
+    InvalidEncoding(secp256k1::Error),
+    #[cfg(feature = "base64")]
+    /// Invalid base64 encoding.
+    InvalidBase64(::base64::DecodeError),
```

Comment:
Feature gating enum variant means that if you match against this enum and another dependency in the tree adds this feature it will break your code because you're no longer being exhaustive

Reply:
That's a good point. Any suggested way to deal with this?

Reply:
Fixed this.

Reply:
> That's a good point. Any suggested way to deal with this?

You can deal with that using `non_exhaustive`, but not in 1.29